### PR TITLE
Update iceberg to 0.13.1

### PR DESCRIPTION
**Pull request by homar**
## Description
Updates apache iceberg version to 0.13.1
## Documentation

(x) No documentation is needed.
( ) Sufficient documentation is included in this PR.
( ) Documentation PR is available with #prnumber.
( ) Documentation issue #issuenumber is filed, and can be handled later.

## Release notes

(x) No release notes entries required.
( ) Release notes entries required with the following suggested text:

```
# Section
* Fix some things. ({issue}`5678`)
```

### CODE REVIEW

**Review by losipiuk — APPROVED** [review]

File: plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergOrcFileWriter.java [hunk 1]
```diff
@@ -104,7 +104,8 @@ public Metrics getMetrics()
     private static Metrics computeMetrics(MetricsConfig metricsConfig, Schema icebergSchema, ColumnMetadata<OrcType> orcColumns, long fileRowCount, Optional<ColumnMetadata<ColumnStatistics>> columnStatistics)
     {
         if (columnStatistics.isEmpty()) {
-            return new Metrics(fileRowCount, null, null, null, null, null);
+            // TODO add implementation of Map<Integer, Long> nanValueCounts
```
Comment (findepi) [inline plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergOrcFileWriter.java:107]: Make it a Note, not TODO. CBO has no use for NaN counts today

File: plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/TestMetricsWrapper.java [hunk 2]
```diff
@@ -46,7 +46,7 @@ public void testRoundTrip()
         Map<Integer, ByteBuffer> lowerBounds = ImmutableMap.of(13, ByteBuffer.wrap(new byte[] {0, 8, 9}));
         Map<Integer, ByteBuffer> upperBounds = ImmutableMap.of(17, ByteBuffer.wrap(new byte[] {5, 4, 0}));
 
-        Metrics expected = new Metrics(recordCount, columnSizes, valueCounts, nullValueCounts, lowerBounds, upperBounds);
+        Metrics expected = new Metrics(recordCount, columnSizes, valueCounts, nullValueCounts, null, lowerBounds, upperBounds);
```
Comment (findepi) [inline plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/TestMetricsWrapper.java:49]: empty map instead?

**Review by findepi — APPROVED** [review]

**Comment by findepi** [comment]
@homar are you aware of any changes impacting users that flows from this upgrade?

**Comment by homar** [comment]
> @homar are you aware of any changes impacting users that flows from this upgrade?

I am not.  Main reason for this update is that it should allow us to implement row level deletes and updates for orc files.

**Review by losipiuk — APPROVED** [review]

File: plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/MetricsWrapper.java [hunk 3]
```diff
@@ -40,6 +41,7 @@ public MetricsWrapper(
                 columnSizes,
                 valueCounts,
                 nullValueCounts,
+                ImmutableMap.of(), // Trino CBO doesn't utilise nanValueCounts at all
```
Comment (electrum) [inline plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/MetricsWrapper.java:44]: These are Iceberg metrics that are part of the Iceberg specification, and thus we should be writing them. The purpose of `MetricsWrapper` is to allow JSON serialization of Iceberg `Metrics`, so we should be pass the value through here. The Parquet writer already collects NaN counts, and the ORC writer should be updated to do so.

Saying "Trino doesn't use them so we don't care" is hostile.

File: plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergOrcFileWriter.java [hunk 4]
```diff
@@ -104,7 +104,8 @@ public Metrics getMetrics()
     private static Metrics computeMetrics(MetricsConfig metricsConfig, Schema icebergSchema, ColumnMetadata<OrcType> orcColumns, long fileRowCount, Optional<ColumnMetadata<ColumnStatistics>> columnStatistics)
     {
         if (columnStatistics.isEmpty()) {
-            return new Metrics(fileRowCount, null, null, null, null, null);
+            // Trino CBO doesn't utilise nanValueCounts at all
```
Comment (electrum) [inline plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergOrcFileWriter.java:107]: Remove comment

File: plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergOrcFileWriter.java [hunk 5]
```diff
@@ -153,6 +154,7 @@ private static Metrics computeMetrics(MetricsConfig metricsConfig, Schema iceber
                 null, // TODO: Add column size accounting to ORC column writers
                 valueCounts.isEmpty() ? null : valueCounts,
                 nullCounts.isEmpty() ? null : nullCounts,
+                null, // Trino CBO doesn't utilise nanValueCounts at all
```
Comment (electrum) [inline plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergOrcFileWriter.java:157]: Replace comment with
```java
// TODO: Add nanValueCounts to ORC writer

**Review by electrum — CHANGES_REQUESTED** [review]

**Review by losipiuk — APPROVED** [review]

**Comment by homar** [comment]
@electrum could you please take another look ?

**Comment by alexjo2144** [comment]
One of the changes in 0.13 was to deprecate the `write.folder-storage.path` and `write.object-storage.path` properties, replacing them with `write.data.path`
https://github.com/apache/iceberg/pull/3094

Can you add an appropriate test to `TestIcebergSparkCompatibility` similar to the existing one for the deprecated properties?
https://github.com/trinodb/trino/blob/master/testing/trino-product-tests/src/main/java/io/trino/tests/product/iceberg/TestIcebergSparkCompatibility.java#L822

File: plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/TestIcebergOrcMetricsCollection.java [hunk 6]
```diff
@@ -237,7 +237,7 @@ public void testBasic()
 
         // Check NaN value count
         // TODO: add more checks after NaN info is collected
-        assertNull(datafile.getNanValueCounts());
+        assertEquals(datafile.getNanValueCounts(), null);
```
Comment (electrum) [inline plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/TestIcebergOrcMetricsCollection.java:240]: Revert

File: plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/TestIcebergOrcMetricsCollection.java [hunk 7]
```diff
@@ -328,7 +328,7 @@ public void testWithNaNs()
         datafile.getValueCounts().values().forEach(valueCount -> assertEquals(valueCount, (Long) 3L));
 
         // TODO: add more checks after NaN info is collected
-        assertNull(datafile.getNanValueCounts());
+        assertEquals(datafile.getNanValueCounts(), null);
```
Comment (electrum) [inline plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/TestIcebergOrcMetricsCollection.java:331]: Revert

File: plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/TestMetricsWrapper.java [hunk 8]
```diff
@@ -46,7 +46,7 @@ public void testRoundTrip()
         Map<Integer, ByteBuffer> lowerBounds = ImmutableMap.of(13, ByteBuffer.wrap(new byte[] {0, 8, 9}));
         Map<Integer, ByteBuffer> upperBounds = ImmutableMap.of(17, ByteBuffer.wrap(new byte[] {5, 4, 0}));
 
-        Metrics expected = new Metrics(recordCount, columnSizes, valueCounts, nullValueCounts, lowerBounds, upperBounds);
+        Metrics expected = new Metrics(recordCount, columnSizes, valueCounts, nullValueCounts, ImmutableMap.of(), lowerBounds, upperBounds);
```
Comment (electrum) [inline plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/TestMetricsWrapper.java:49]: Add values so we test the serialization

Comment (electrum) [reply plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/TestMetricsWrapper.java:49]: We should also test NaN counts for Parquet

**Review by electrum — APPROVED** [review]
Minor comments, otherwise looks good

**Comment by findinpath** [comment]
In case that this PR gets merged only after https://github.com/trinodb/trino/pull/11062 please do remove the Iceberg related code for deleting table metadata files  in `io.trino.plugin.iceberg.TrinoHiveCatalog#dropTable`

File: plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/TestMetricsWrapper.java (hunk 8 above)
Comment (homar) [reply plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/TestMetricsWrapper.java:49]: I don't see any tests for metrics for Parquet format. Nothing like `TestIcebergOrcMetricsCollection` exists for parquet. I will add a ticket to create such a test

Comment (homar) [reply plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/TestMetricsWrapper.java]: https://github.com/trinodb/trino/issues/11182

**Pull request merged**